C/Cpp_include/c4Query.hh

```diff
@@ -134,22 +134,22 @@ struct C4QueryObserver
     : public fleece::InstanceCounted
     , C4Base {
   public:
-    virtual ~C4QueryObserver() = default;
+    ~C4QueryObserver() override = default;
 
-    C4Query* query() const { return _query; }
+    [[nodiscard]] C4Query* query() const { return _query; }
 
     virtual void setEnabled(bool enabled) = 0;
 
     /// If the latest run of the query failed, the error will be stored here, with nonzero `code`.
     /// Always check the error before getting the enumerator.
-    C4Error getError() const { return _currentError; }
+    [[nodiscard]] C4Error getError() const { return _currentError; }
 
     /// Returns a new enumerator on the query results.
     /// If the query failed, throws that error as an exception.
-    virtual C4Query::Enumerator getEnumerator(bool forget = true) = 0;
+    virtual C4Query::Enumerator getEnumerator(bool forget = true) = 0;  // NOLINT(google-default-arguments)
```

Comment:
As we have done this in many places, I'm wondering if Clang-tidy has a config file that we can ignore this rule by default?  What is your opinion on this?

Reply:
My opinion is there's probably a better way to do this than the way we do it, but it would be a tremendous change to fix it. So in the meantime, yes we could create a `.clang-tidy` file I think in order to disable this warning.